oak_docker_linux_init/README.md

```diff
@@ -7,46 +7,174 @@
+the directory, which launches the docker entry point after setting up the
+necessary environment. You can use this `init` binary as the `/init` for the
+initramfs or as the initial binary when you `chroot` to the filesystem extracted
+from the Docker image.
+
+### Docker filesystem as a chroot tree
+
+If the Docker image requires drivers and additional setup, we can have custom
+Docker launcher that is packaged as an initramfs image and then launch the
+docker container. We will use an [Alpine Linux](https://www.alpinelinux.org/)
+distribution for the purposes of this prototype. Ideally, we want to implement
+the Docker launcher from scratch and build a custom Linux kernel with all
+drivers, but the Alpine Linux distribution will help illustrate and derisk the
+changes needed to support Docker on top of Oak stage0.
+
+We will need the following packages from
+[Alpine Linux Downloads](https://www.alpinelinux.org/downloads/) Page:
+
+- [Alpine Minirootfs](https://dl-cdn.alpinelinux.org/alpine/v3.17/releases/x86_64/alpine-minirootfs-3.17.3-x86_64.tar.gz):
+  for an initial filesystem.
+- [Alpine Netboot](https://dl-cdn.alpinelinux.org/alpine/v3.17/releases/x86_64/alpine-netboot-3.17.3-x86_64.tar.gz):
+  for a Linux kernel and drivers.
+
+Download the necesary files:
+
+```bash
+OAK_DOCKER_PREP_DIR=$(mktemp -d)
+MINIROOTFS_TGZ="${OAK_DOCKER_PREP_DIR}/alpine-minirootfs.tar.gz"
+wget https://dl-cdn.alpinelinux.org/alpine/v3.17/releases/x86_64/alpine-minirootfs-3.17.3-x86_64.tar.gz \
+ -O "${MINIROOTFS_TGZ}"
+
+NETBOOT_TGZ="${OAK_DOCKER_PREP_DIR}/alpine-netboot.tar.gz"
+wget https://dl-cdn.alpinelinux.org/alpine/v3.17/releases/x86_64/alpine-netboot-3.17.3-x86_64.tar.gz \
+ -O "${NETBOOT_TGZ}"
+
+EXTRACT_VMLINUX="${OAK_DOCKER_PREP_DIR}/extract-vmlinux.sh"
+wget https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/plain/scripts/extract-vmlinux \
+ -O "${EXTRACT_VMLINUX}"
+
+```
+
+Extract the initramfs from netboot:
+
+```bash
+NETBOOT_INITRAMFS="${OAK_DOCKER_PREP_DIR}/netboot-initramfs"
+tar xzvf "${NETBOOT_TGZ}" boot/initramfs-virt -O > "${NETBOOT_INITRAMFS}"
+```
+
+Also extract the corresponding compressed kernel and extract the vmlinux file.
+You will need the
+[extract-vmlinux](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/plain/scripts/extract-vmlinux)
+script from the kernel source code.
+
+```bash
+NETBOOT_COMPRESSED_KERNEL="${OAK_DOCKER_PREP_DIR}/netboot-vmlinuz"
+NETBOOT_VMLINUX="${OAK_DOCKER_PREP_DIR}/netboot-vmlinux"
+
+tar xzvf "${NETBOOT_TGZ}" boot/vmlinuz-virt -O > "${NETBOOT_COMPRESSED_KERNEL}"
+sh "${EXTRACT_VMLINUX}" "${NETBOOT_COMPRESSED_KERNEL}" > "${NETBOOT_VMLINUX}"
+```
+
+You might want to copy the kernel to the `bin/` at the root of Oak's workspace
+to use as the Linux kernel when launching QEMU later:
+
+```bash
+cp ${NETBOOT_VMLINUX} bin/vmlinux
 ```
 
-Build the Stage 0 Firmware image:
+Note that you need to extract the kernel from so that it will be compatible with
```

Comment:
There seems to be something missing after `from` in the sentence.

Reply:
thanks for catching this. Fixed.